BDArmory/Competition/OrchestrationStrategies/WaypointFollowingStrategy.cs

```diff
@@ -103,7 +103,7 @@ public IEnumerator Execute(BDAScoreClient client, BDAScoreService service)
                 yield return new WaitWhile(() => BDACompetitionMode.Instance.competitionIsActive); //DoUpdate handles the deathmatch half of the combat waypoint race and ends things when only 1 team left
             }
             else
-                yield return new WaitWhile(() => BDACompetitionMode.Instance.competitionIsActive && pilots.Any(pilot => pilot != null && pilot.weaponManager != null && pilot.IsRunningWaypoints && pilot.TakingOff || (!pilot.TakingOff && !(pilot.vessel.Landed || pilot.vessel.Splashed)))); 
+                yield return new WaitWhile(() => pilots.Any(pilot => pilot != null && pilot.weaponManager != null && pilot.IsRunningWaypoints && (pilot.TakingOff || !(pilot.vessel.Landed || pilot.vessel.Splashed)))); 
```

Comment:
Is `!(pilot.vessel.Landed || pilot.vessel.Splashed)` any different from `!pilot.vessel.LandedOrSplashed`?

Also, if non-pilot AIs are allowed to run waypoints, then the (equivalent?) `(pilot.TakingOff || !pilot.vessel.LandedOrSplashed)` check should be inside a check for what type of AI it is and the conditions adjusted appropriately. This will be much easier once I've finished the multi-craft stuff as you can just ask the AI what type it is instead of having to test-cast it to the various types, so maybe this should just be flagged for now and dealt with later.

Reply:
A simple check for now could be something like
```csharp
var checkLandedOrSplashed = pilots.ToDictionary(p => p, p => p as BDModuleSurfaceAI == null);
yield return new WaitWhile(() => pilots.Any(pilot => pilot != null && pilot.weaponManager != null && pilot.IsRunningWaypoints && (pilot.TakingOff || (checkLandedOrSplashed[pilot] && !pilot.vessel.LandedOrSplashed))));
```
if
```csharp
TakingOff = false;
```
is added to `BDModuleSurfaceAI.ActivatePilot` (like in BDModuleOrbitalAI.cs).

Reply:
Why? That seems less optimized than the current? It's a binary check - either the AI is intentionally on the ground(pilot/vtol TakingOff, surface is permanently in this state), and does not check landedOrSplashed, or they've flagged takingOff to false upon getting airborne, and does check landedOrSplashed because it shouldn't be; TakingOff is already returning true, so the landedOrSplashed check is already irrelevant for SrfAIs before creating a Dictionary to iterate through to check for that.